api/src/main/java/org/openmrs/module/metadatamapping/api/db/MetadataMappingDAO.java

```diff
@@ -130,4 +133,115 @@
+	 * @param metadataSetMemberId database id of the object
+	 * @return object or null, if does not exist
+	 */
+	MetadataSetMember getMetadataSetMember(Integer metadataSetMemberId);
+	
+	/**
+	 * Get members of a metadata set. If members have {@link MetadataSetMember#getSortWeight()} set they will be ordered 
+	 * in ascending order according to said weight. Note that due to differences in database implementations, the order 
+	 * will be unpredictable, if there are null sort weights in the set.
+	 * @param metadataSet metadata set
+	 * @param firstResult zero based index of first result to get 
+	 * @param maxResults maximum number of results to get
+	 * @param retiredHandlingMode handle retired objects using this mode
+	 * @return list of members in the order defined by the optional {@link MetadataSetMember#getSortWeight()} values
+	 * @see #getMetadataSetMembers(String, String, int, int, RetiredHandlingMode)
+	 */
+	List<MetadataSetMember> getMetadataSetMembers(MetadataSet metadataSet, int firstResult, int maxResults,
+	        RetiredHandlingMode retiredHandlingMode);
+	
+	/**
+	 * Get members of a metadata set. If members have {@link MetadataSetMember#getSortWeight()} set they will be ordered 
+	 * in ascending order according to said weight. Note that due to differences in database implementations, the order 
+	 * will be unpredictable, if there are null sort weights in the set.
+	 * @param metadataSourceName name of the source of the set
+	 * @param metadataSetCode code of the set
+	 * @param firstResult zero based index of first result to get 
+	 * @param maxResults maximum number of results to get
+	 * @param retiredHandlingMode handle retired objects using this mode
+	 * @return list of members in the order defined by the optional {@link MetadataSetMember#getSortWeight()} values
+	 * @see #getMetadataSetMembers(MetadataSet, int, int, RetiredHandlingMode)
+	 */
+	List<MetadataSetMember> getMetadataSetMembers(String metadataSourceName, String metadataSetCode, int firstResult,
+	        int maxResults, RetiredHandlingMode retiredHandlingMode);
+	
+	/**
+	 * Get unretired metadata items in the set. If set members have {@link MetadataSetMember#getSortWeight()} set they will 
+	 * be ordered in ascending order according to said weight. Note that due to differences in database implementations, 
+	 * the order  will be unpredictable, if there are null sort weights in the set.
+	 * @param type type of the metadata items
+	 * @param metadataSet metadata set
+	 * @param firstResult zero based index of first result to get 
+	 * @param maxResults maximum number of results to get
+	 * @param <T> type of the metadata items
+	 * @return list of items in the order defined by the optional {@link MetadataSetMember#getSortWeight()} values
+	 * @see #getMetadataSetItems(Class, String, String, int, int)
+	 */
+	<T extends OpenmrsMetadata> List<T> getMetadataSetItems(Class<T> type, MetadataSet metadataSet, int firstResult,
```

Comment:
Looking back at the original design without MetadataSetMember it would allow for that. I wonder if it was intentional.